test/test-parse-statement.cpp

```diff
@@ -715,6 +715,25 @@ TEST(test_parse, else_without_if) {
   }
 }
 
+TEST(test_parse, missing_if_after_else) {
+  {
+    spy_visitor v;
+    padded_string code(u8"if (false) {} else (true) {}"_sv);
```

Comment:
Nit: add test cases:

* should *not* report for `if (false) {} else true {}` (it should complain about `{` instead)
* should *not* report anything for `if (false) {} else (true)NEWLINE{}`